### PR TITLE
fix(tanstack-table): reset page to 1 when change page size

### PR DESCRIPTION
**Pull request by H4ad**
## Pull Request

---

Fixes the issue of page not being reset to 1 after page size change.

https://github.com/user-attachments/assets/44b53716-f7bb-4b94-a93c-0278e039150a

Fixes https://github.com/SigNoz/engineering-pod/issues/4926

---

### ✅ Change Type
_Select all that apply_

- [ ] ✨ Feature
- [x] 🐛 Bug fix
- [ ] ♻️ Refactor
- [ ] 🛠️ Infra / Tooling
- [ ] 🧪 Test-only

---

- Tests added/updated: Yes
- Manual verification: Yes
- Edge cases covered: -

---

- Blast radius: Tanstack Table
- Potential regressions: None
- Rollback plan: Revert this commit

---

| Field | Value |
|------|-------|
| Deployment Type | Cloud / OSS / Enterprise |
| Change Type | Bug Fix |
| Description | The tables of Infrastructure Monitoring were not reseting to page 1 after changing the page size, now this behavior was fixed. |

---

### 📋 Checklist
- [x] Tests added or explicitly not required
- [x] Manually tested
- [ ] Breaking changes documented
- [ ] Backward compatibility considered

### CODE REVIEW

**Review by AshwinBhatkal — DISMISSED** [review]

File: frontend/src/components/TanStackTableView/TanStackTable.tsx [hunk 1]
```diff
@@ -625,6 +625,7 @@ function TanStackTableInner<TData>(
 										defaultValue="10"
 										onChange={(value): void => {
 											setLimit(+value);
+											setPage(1);
```
Comment (ahrefabhi) [inline frontend/src/components/TanStackTableView/TanStackTable.tsx]: we should call pagination.onPageChange?.(1) after this

Comment (H4ad) [reply frontend/src/components/TanStackTableView/TanStackTable.tsx]: Yes, great catch

**Review by ahrefabhi — APPROVED** [review]

**Review by AshwinBhatkal — APPROVED** [review]
